### ISSUE TITLE
Tests failing locally for a number of det/michigan tests due to ordering issues

### ISSUE THREAD

**Issue by bonfirefan** [City-Bureau/city-scrapers#484]
Working with @mkrump on this - if anyone else can replicate this failure, this occurs when running tests locally. Occurs with `pytest -s tests` as well. Appears to be due to items being referenced from unordered dictionaries .

When the list of parsed items are ordered, this resolves the error. 
```
=================================== FAILURES ===================================
_______________________________ test_prev_start ________________________________

    def test_prev_start():
>       assert parsed_prev_items[0]['start'] == {
            'date': date(2017, 6, 28), 'time': None, 'note': ''
        }
E       AssertionError: assert {'date': date... 'time': None} == {'date': datet... 'time': None}
E         Omitting 2 identical items, use -vv to show
E         Differing items:
E         {'date': datetime.date(2016, 9, 14)} != {'date': datetime.date(2017, 6, 28)}
E         Use -v to get the full diff

tests/test_det_downtown_development_authority.py:132: AssertionError
_________________________________ test_prev_id _________________________________

    def test_prev_id():
>       assert parsed_prev_items[0]['id'] == \
               'det_downtown_development_authority/201706280000/x/board_of_directors'
E       AssertionError: assert 'det_downtown..._of_directors' == 'det_downtown_..._of_directors'
E         - det_downtown_development_authority/201609140000/x/board_of_directors
E         ?                                        ^^^^
E         + det_downtown_development_authority/201706280000/x/board_of_directors
E         ?                                       ++ ^^

tests/test_det_downtown_development_authority.py:144: AssertionError
_____________________________ test_prev_documents ______________________________

    def test_prev_documents():
>       assert parsed_prev_items[0]['documents'] == [
            {
                'url': 'http://www.degc.org/wp-content/uploads/2017-06-28-dda-special-board-meeting-materials.pdf',
                'note': 'meeting materials',
            },
            {
                'url': 'http://www.degc.org/wp-content/uploads/2017-06-28-Attachment-D.pdf',
                'note': 'attachment d',
            },
            {
                'url': 'http://www.degc.org/wp-content/uploads/2017-06-28-Attachment-K.pdf',
                'note': 'attachment k',
            },
            {
                'url': 'http://www.degc.org/wp-content/uploads/2017-06-28-DDA-Board-Meeting-Minutes.pdf',
                'note': 'minutes',
            },
        ]
E       AssertionError: assert [{'note': 'ag...Minutes.pdf'}] == [{'note': 'mee...Minutes.pdf'}]
E         At index 0 diff: {'note': 'agenda', 'url': 'http://www.degc.org/wp-content/uploads/2016-09-14-DDA-Board-Meeting-Agenda.pdf'} != {'note': 'meeting materials', 'url': 'http://www.degc.org/wp-content/uploads/2017-06-28-dda-special-board-meeting-materials.pdf'}
E         Right contains more items, first extra item: {'note': 'attachment k', 'url': 'http://www.degc.org/wp-content/uploads/2017-06-28-Attachment-K.pdf'}
E         Use -v to get the full diff

tests/test_det_downtown_development_authority.py:163: AssertionError
_______________________________ test_prev_start ________________________________

    def test_prev_start():
>       assert parsed_prev_items[0]['start'] == {
            'date': date(2018, 6, 19), 'time': None, 'note': ''
        }
E       AssertionError: assert {'date': date... 'time': None} == {'date': datet... 'time': None}
E         Omitting 2 identical items, use -vv to show
E         Differing items:
E         {'date': datetime.date(2017, 12, 12)} != {'date': datetime.date(2018, 6, 19)}
E         Use -v to get the full diff

tests/test_det_local_development_finance_authority.py:129: AssertionError
_________________________________ test_prev_id _________________________________

    def test_prev_id():
>       assert parsed_prev_items[0]['id'] \
               == 'det_local_development_finance_authority/201806190000/x/board_of_directors'
E       AssertionError: assert 'det_local_de..._of_directors' == 'det_local_dev..._of_directors'
E         Skipping 33 identical leading characters in diff, use -v to show
E         - hority/201712120000/x/board_of_directors
E         ?           ^ ^^^
E         + hority/201806190000/x/board_of_directors
E         ?           ^^^ ^

tests/test_det_local_development_finance_authority.py:141: AssertionError
_____________________________ test_prev_documents ______________________________

    def test_prev_documents():
>       assert parsed_prev_items[1]['documents'] == [
            {
                'url': 'http://www.degc.org/wp-content/uploads/12-12-17-LDFA-Board-Meeting-Minutes.pdf',
                'note': 'minutes',
            },
            {
                'url': 'http://www.degc.org/wp-content/uploads/12-12-17-Special-LDFA-Board-Meeting-Agenda-Only.pdf',
                'note': 'agenda',
            },
        ]
E       AssertionError: assert [{'note': 'ag...da-Only.pdf'}] == [{'note': 'min...da-Only.pdf'}]
E         At index 0 diff: {'note': 'agenda', 'url': 'http://www.degc.org/wp-content/uploads/06-19-18-Special-LDFA-Board-Meeting-Agenda-Only.pdf'} != {'note': 'minutes', 'url': 'http://www.degc.org/wp-content/uploads/12-12-17-LDFA-Board-Meeting-Minutes.pdf'}
E         Right contains more items, first extra item: {'note': 'agenda', 'url': 'http://www.degc.org/wp-content/uploads/12-12-17-Special-LDFA-Board-Meeting-Agenda-Only.pdf'}
E         Use -v to get the full diff

tests/test_det_local_development_finance_authority.py:160: AssertionError
_______________________________ test_prev_start ________________________________

    def test_prev_start():
>       assert parsed_prev_items[0]['start'] == {
            'date': date(2017, 6, 27), 'time': None, 'note': ''
        }
E       AssertionError: assert {'date': date... 'time': None} == {'date': datet... 'time': None}
E         Omitting 2 identical items, use -vv to show
E         Differing items:
E         {'date': datetime.date(2017, 8, 8)} != {'date': datetime.date(2017, 6, 27)}
E         Use -v to get the full diff

tests/test_next_michigan_development_corporation.py:127: AssertionError
_________________________________ test_prev_id _________________________________

    def test_prev_id():
>       assert parsed_prev_items[0]['id'] \
               == 'next_michigan_development_corporation/201706270000/x/board_of_directors'
E       AssertionError: assert 'next_michiga..._of_directors' == 'next_michigan..._of_directors'
E         Skipping 33 identical leading characters in diff, use -v to show
E         - tion/201708080000/x/board_of_directors
E         ?           ^^^
E         + tion/201706270000/x/board_of_directors
E         ?           ^^^

tests/test_next_michigan_development_corporation.py:139: AssertionError
_____________________________ test_prev_documents ______________________________

    def test_prev_documents():
>       assert parsed_prev_items[0]['documents'] == [
            {
                'url': 'http://www.degc.org/wp-content/uploads/06-27-17-DNMDC-Board-Meeting-Minutes.pdf',
                'note': 'minutes',
            },
            {
                'url': 'http://www.degc.org/wp-content/uploads/2017-06-27-DNMDC-Special-Board-Meeting-Agenda-Only-1.pdf',
                'note': 'agenda',
            },
        ]
E       AssertionError: assert [{'note': 'ag...nda-4-1.pdf'}] == [{'note': 'min...-Only-1.pdf'}]
E         At index 0 diff: {'note': 'agenda', 'url': 'http://www.degc.org/wp-content/uploads/2016-08-09-DNMDC-Special-Board-Meeting-Agenda-4-1.pdf'} != {'note': 'minutes', 'url': 'http://www.degc.org/wp-content/uploads/06-27-17-DNMDC-Board-Meeting-Minutes.pdf'}
E         Right contains more items, first extra item: {'note': 'agenda', 'url': 'http://www.degc.org/wp-content/uploads/2017-06-27-DNMDC-Special-Board-Meeting-Agenda-Only-1.pdf'}
E         Use -v to get the full diff

tests/test_next_michigan_development_corporation.py:158: AssertionError
==================== 9 failed, 6629 passed in 12.93 seconds ====================

```

**Comment by mkrump** [comment]
Try #486 when you get a chance to see if this resolves your issue.

**Comment by mkrump** [comment]
Merged #486. Feel free to reopen if this doesn't resolve the issue.
